beacon-chain/blockchain/service.go

```diff
@@ -63,14 +65,25 @@ func NewChainService(ctx context.Context, cfg *Config) (*ChainService, error) {
 		canonicalCrystallizedStateFeed: new(event.Feed),
 		blocksPendingProcessing:        [][32]byte{},
 		devMode:                        cfg.DevMode,
+		slotAlignmentDuration:          params.SlotDuration,
 	}, nil
 }
 
 // Start a blockchain service's main event loop.
 func (c *ChainService) Start() {
 	// TODO(#474): Fetch the slot: (block, state) DAGs from persistent storage
 	// to truly continue across sessions.
-	log.Infof("Starting service")
+	log.Info("Starting service")
+
+	// If the genesis time was at 12:00:00PM and the current time is 12:00:03PM,
+	// the next slot should tick at 12:00:08PM. We can accomplish this
+	// using utils.WaitUntilTimestamp and passing in the desired
+	// slot duration.
+	//
+	// Instead of utilizing params.SlotDuration, we utilize a property of
+	// RPC service struct so this value can be set to 0 seconds
+	// as a parameter in tests. Otherwise, tests would sleep.
+	utils.WaitUntilTimestamp(time.Duration(c.slotAlignmentDuration) * time.Second)
```

Comment:
I don't quite get this... Do we always wait 8 seconds before we start the NewTicker? And should this be inside `updateHead` similar to `ValidatorAssignment`?

Reply:
We don't wait 8 seconds, we wait until the timestamp is a multiple of 8 seconds. So if the current slot started at 12:00:00 and the current time is 12:00:03, we wait until 12:00:08 to start ticking.

Reply:
Got it!

Reply:
Can this be `utils.BlockingWait`? There is no "timestamp" to speak of and the only parameter is a time.Duration.